### DISCUSSION TITLE
[2023-05-30] Library and Engine Team Update

### DISCUSSION THREAD

**Discussion by jdunkerley** [enso-org/enso#6896] [Weekly Demo Day]
- James: Table write control only writes first 1000 rows to temporary file and made dry run stuff reusable.
- James: Fix for Regex match iterator so doesn't infinite loop.
- James: Adding ALIASes and drop downs to various APIs.
- Radek: Implemented support for casting a `Mixed` column. #6777
- Radek: Added `add_row_number` to in-memory `Table`. #6890 
- Greg: Added rounding functions to in-memory `Column`. #6817 
- Jaroslav: Various test improvements to increase coverage and speed things up. #6820 
- Jaroslav: Infrastructure for testing inter projects imports/exports. #6840 
- Jaroslav: Runtime checking of ascribed types. #6790 
- Hubert: Fix issue where 2 renames were causing a problem at exit. #6842 
- Hubert: Append warnings on tail calls. #6849 
- Hubert: Empty editions not parsed as NaN anymore. #6873 
- Hubert: Investigating double-clicking problem. #6763
- Hubert: Investigating Invalid Version mismatch problems. #6701
- Pavel: Upgrade to GraalVM 22.3.1 with JDK17. #6750 
- Pavel: Allow importing polyglot java inner classes. #6818
- Pavel: Add all the classes to build-time initialization for native-image. #6879
- Pavel: Added compiler pass to analyze non-existing imported symbols. #6726
- Pavel: Added compiler pass to detect ambiguous imports. #6868
- Dmitry: Consistent self type names, resolves issues with static methods. #6867

**Comment by radeusgd** [comment]
#### `add_row_number`:

![image](https://github.com/enso-org/enso/assets/1436948/ae8d96e7-e890-45b4-aef5-5958e5e13a90)

**Comment by radeusgd** [comment]
#### Mixed `cast`

![image](https://github.com/enso-org/enso/assets/1436948/0add2fce-d63f-4204-8c3d-32e244a13e6a)
